src/gui/gt_mdiitem.cpp

```diff
@@ -101,6 +135,12 @@ GtMdiItem::isPrintable() const
     return false;
 }
 
+std::vector<GtQmlAction *>
+GtMdiItem::toolbarActions() const
+{
+    return pimpl->m_toolbarActions;
+}
```

Comment:
Since the function is const, shouldn't this return a vector with const elements? Also, I am unsure whether we should use `std::vector` or `QVector` here and whether we should return it as a copy or reference (A `QVector` should be cheaper to shallow copy?). 

For such use cases it would be neat to return a more abstract range-like object, like a span just that it does not need to care about the underlying container. That does not heap allocate on copy and is also ABI safe-ish. But I might very well be overthinking this 😂 This function might very well be not called that often anyways. 
```suggestion
std::vector<const GtQmlAction*>
GtMdiItem::toolbarActions() const
{
    return gt::const_container_cast(pimpl->m_toolbarActions);
}
```

Reply:
The non-constness is by design. Since the toolbar potentially interacts in a non-const way with the actions. Slots are called and so on.

Reply:
In fact, we could simply return it as a const reference. The action vector is already part of the mdi item.

Reply:
Ah I see